### PR TITLE
0.4

### PR DESCRIPTION
**Pull request by riverar**
Simplifies code generation and configures sample to be easier to use out of the box.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ extern crate windows;
 
 pub mod helpers;
 
-#[allow(non_camel_case_types, non_snake_case, non_upper_case_globals, clippy::derivable_impls, clippy::missing_safety_doc, clippy::too_many_arguments)]
+#[allow(non_camel_case_types, non_snake_case, non_upper_case_globals, clippy::useless_transmute, clippy::extra_unused_lifetimes, clippy::derivable_impls, clippy::missing_safety_doc, clippy::too_many_arguments)]
 mod Microsoft;
 
 pub use Microsoft::Dia::*;
```
Comment (kennykerr) [inline src/lib.rs:10]: Any reason we need this layer of modules?

File: rustfmt.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 newline_style = "Unix"
-max_width = 500
+max_width = 500
```
Comment (kennykerr) [inline rustfmt.toml:2]: Why is this needed?

File: crates/tools/api/src/main.rs [hunk 3]
```diff
@@ -1,56 +1,44 @@
...
-    let _ = std::fs::remove_dir_all(&output_path);
-    output_path.pop();
+    let output_path = PathBuf::from("src/Microsoft/Dia");
+    fs::remove_dir_all(&output_path).unwrap();
+    fs::create_dir_all(&output_path).unwrap();
 
-    let winmd_files = [File::new(".windows/winmd/Microsoft.Dia.winmd").unwrap(), File::new(".windows/winmd/Windows.Win32.winmd").unwrap(), File::new(".windows/winmd/Windows.Win32.Interop.winmd").unwrap()];
-    let reader = Reader::new(&winmd_files);
-    let root = reader.tree("Microsoft", &[]).expect("Microsoft namespace not found");
+    let mut generated_tokens = windows_bindgen::component("Microsoft.Dia", &files);
```
Comment (kennykerr) [inline crates/tools/api/src/main.rs:20]: This could conceivably just go directly into a `src/bindingsrs`

Comment (riverar) [reply crates/tools/api/src/main.rs:20]: And maintain the `Microsoft::Dia` namespace? (See also https://github.com/microsoft/dia-rs/pull/10#discussion_r1042587972)

Comment (riverar) [reply crates/tools/api/src/main.rs:20]: I see now, will simplify that, thanks!

File: src/lib.rs (hunk 1 above)
Comment (riverar) [reply src/lib.rs:10]: I see now, will simplify that, thanks!

File: rustfmt.toml (hunk 2 above)
Comment (riverar) [reply rustfmt.toml]: Deleted them all.
